support/routes/v2/isobot/ops/botService.ts

```diff
@@ -91,6 +170,276 @@ class BotService {
+      )
+      response.push(
+        `-   A records checker: https://dnschecker.org/#A/${domain}`
+      )
+      response.push(
+        `-   CNAME records checker: https://dnschecker.org/#CNAME/${domain}`
+      )
+    }
+
+    response.push(
+      `-   \`${domain}\` points to ${
+        cnameRecord ? `\`${cnameRecord}\`` : "no CNAME records"
+      }`
+    )
+
+    if (cnameRecord || indirectionDomain) {
+      response.push(
+        `-   \`${cnameRecord || indirectionDomain}\` points to ${
+          intermediateRecords
+            ? `\`${intermediateRecords.join(", ")}\``
+            : "nothing"
+        }`
+      )
+    }
+
+    if (redirectionDomain !== cnameDomain) {
+      response.push(
+        `-   \`${redirectionDomain}\` points to ${
+          redirectionRecords
+            ? `\`${redirectionRecords.join(", ")}\``
+            : "no A records"
+        }`
+      )
+    }
+
+    return this.getSlackMessage(response)
+  }
+
+  dnsChecker(payload: SlackPayload) {
+    // Step 1: Get the domain name provided by the user
+    const { user_name: user, channel_name: channel, text: domain } = payload
+    logger.info({
+      message: "DNS check requested",
+      meta: {
+        method: "dnsChecker",
+        user,
+        channel,
+        domain,
+      },
+    })
+
+    return this.checkCname(domain)
+      .andThen((cname) => {
+        // Original domain does not have a CNAME record, check if the www
+        // version has a valid CNAME record
+        if (!cname && !domain.startsWith("www.")) {
+          const cnameDomain = `www.${domain}`
+          return ResultAsync.combine([
+            okAsync(cnameDomain),
+            this.checkCname(cnameDomain),
+          ])
+        }
+
+        return ResultAsync.combine([okAsync(domain), okAsync(cname)])
+      })
+      .andThen(([cnameDomain, cnameRecord]) => {
+        // Original and www version of the domain do not have a CNAME record,
+        // check if our indirection domain is still correct
+        if (!cnameRecord) {
+          const indirectionDomain = domain.startsWith("www.")
+            ? `${domain
+                .slice(4)
+                .replaceAll(".", "-")}.${DNS_INDIRECTION_DOMAIN}`
+            : `${domain.replaceAll(".", "-")}.${DNS_INDIRECTION_DOMAIN}`
+
+          return ResultAsync.combine([
+            okAsync(null),
+            okAsync(cnameRecord),
+            okAsync(indirectionDomain),
+            this.checkA(indirectionDomain),
+          ])
+        }
+
+        // Either the original or www version of the domain has a CNAME record,
+        // check if the CNAME record is valid
+        return ResultAsync.combine([
+          okAsync(cnameDomain),
+          okAsync(cnameRecord),
+          okAsync(cnameRecord),
+          this.checkA(cnameRecord),
+        ])
+      })
+      .andThen(
+        ([cnameDomain, cnameRecord, indirectionDomain, indirectionRecords]) => {
```

Comment:
nit: prefer obj over passing in an array so we dont mix up the elements accidently

Reply:
No strong opinions here but was thinking of structuring it this way so that each step of the checking process is separate. Alternatively doing the object method means they would be in a single `ResultAsync.combine()`. Possible but would prefer keeping it the current way for better understanding of the process flow, let me know your thoughts.

Reply:
Adjusted in d33c2d0a!